src/AutoMerge/dependency_confusion.jl

```diff
@@ -0,0 +1,50 @@
+# TODO: Add a more thorough explanation of the dependency confusion
+# vulnerability and how this guideline mitigates it.
+
+const guideline_dependency_confusion =
+    Guideline("No UUID conflict with other registries.",
+              data -> has_no_dependency_confusion(data.pkg,
+                                                  data.registry_head,
+                                                  data.public_registries))
+
+# TODO: Needs a strategy to handle connection failures for the public
+# registries. Preferably they should also be cloned only once and then
+# just updated to mitigate the effect of them being temporarily
+# offline. This could be implemented with the help of the Scratch
+# package, but requires Julia >= 1.5.
```

Comment:
I'd be in favor of bumping the minimum version to Julia 1.5 if it helps here; the syntax benefits would be nice too (no more `x=x` in keyword arguments). And since RegistryCI is usually run by itself in CI (and is a leaf of the dependency tree), one doesn't need to upgrade their whole stack to use the new version.

Reply:
a related concern is if an external registry gets corrupted and has an invalid TOML file, say. Then RegistryCI would throw everytime it gets to this check. I wonder if we should use `TOML.tryparsefile` and fail the check if we get an error, so at least RegistryCI doesn't crash in that event, but we just fail this check each time. That would still be pretty disruptive though.

Reply:
It can definitely be done with Scratch, but maybe that's a secondary concern.

Invalid TOML files or other registry irregularities is a real risk. The easiest protection would be a `try/catch`, but that's rather heavyhanded.

Reply:
I think a try/catch around that would be OK; it seems like that code should only fail if there are issues cloning and parsing the registry. And if we fail the check in the event of an error, we won't be sweeping any issues under the rug, since automerge will fail.

Reply:
It's slightly unfair to the affected PRs since it's in no way their fault (well, unless they have made a DOS attack against the registry to try to disable the check...), but I agree that getting attention to the issue is of higher value.

Does someone know how  `clone_repo` behaves when there are network problems?

Reply:
yeah, agreed it's not great for the affected PRs... it seems too risky to pass the check if there's an error though, since then an attacker could even do something like add a new public registry they control, then invalidate the TOML or cause network issues in that registry, and then get their package in during the auto-pass window before the problem is fixed.

Reply:
I've added a `try/catch` with an explanatory message.